test/mkmf/base.rb

```diff
@@ -147,7 +147,10 @@ def config_value(name)
 
   include Base
 
-  def assert_separately(args, src, *rest, **options)
-    super(args + ["-r#{__FILE__}"], "extend TestMkmf::Base; setup\nEND{teardown}\n#{src}", *rest, **options)
+  def assert_separately(args, extra_args, src, *rest, **options)
+    super(args + ["-r#{__FILE__}"] + %w[- --] + extra_args,
```

Comment:
In this part, I wanted to write `args + %w[-r#{__FILE__} - --] + extra_args`. However, I saw the `#{__FILE__}` was not converted to the value. So, the `["-r#{__FILE__}"] + %w[- --]` is a bit compromised way.

```
$ ruby -v
ruby 3.3.0dev (2023-05-25T12:45:21Z wip/mkmf-verbose-r.. cbeba2ed16) [x86_64-linux]

$ which irb
~/.local/ruby-cbeba2ed16/bin/irb

$ irb
irb(main):001:0> %w[-r#{__FILE__} - --]
=> ["-r\#{__FILE__}", "-", "--"]

irb(main):002:0> %[-r#{__FILE__} - --]
=> "-r(irb) - --"
```